subsys/ipc/ipc_service/backends/ipc_rpmsg_multi_core.c

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2021, Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ipc/ipc_service_backend.h>
+#include <ipc/rpmsg_multi_instance.h>
+
+#include <logging/log.h>
+#include <sys/util_macro.h>
+#include <zephyr.h>
+#include <device.h>
+
+#define DT_DRV_COMPAT ipc_rpmsg_mc
+
+LOG_MODULE_REGISTER(ipc_rpmsg_multi_instance, CONFIG_IPC_SERVICE_LOG_LEVEL);
+
+#define NUM_INSTANCES		DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT)
+#define NUM_ENDPOINTS		CONFIG_IPC_BACKEND_RPMSG_NUM_ENDPOINTS_PER_INSTANCE
```

Comment:
I think number of endpoint could be added to devicetree node as a new property, to allow variable number of endpoints between instances.